### PR TITLE
[Cleanup] Additional Save Options | Recurring Invoice

### PR DESCRIPTION
**Pull request by Civolilah**
@beganovich @turbo124 PR includes adding new save options to the `Recurring Invoice` creation and editing page to avoid two steps for sending and starting invoice. I've tried several solutions and this one makes the most sense for me, and this one is much more customizable than the others.

- We have a solution to put additional actions in the `More Actions` dropdown, but I haven't implemented it because I'm not sure it will be good for UX. I don't think the user will ever open that extra drop down menu to see if there is an extra way to save the invoice.

- Also, I tried the solution from the Flutter app with all separate buttons. But it will create a problem with responsiveness, there will be many more buttons to fit on a small screen. Screenshot:

Screenshot 2023-04-02 at 01 52 48

There is a solution to keep only the `Save Draft CTA` on small screens, but include everything on large screens. Let me know your thoughts about this one.

- IMO, the best solution for us here is to implement something very similar to the solution for expense categories and other entities where we have an additional `Save & Create` action. So, I implemented it here. We have two additional CTAs `Save & Send` and `Save & Start`. On the edit page, `Save & Send` is only available if the invoice is in Draft status. Screenshots:

Screenshot 2023-04-02 at 03 10 14

Screenshot 2023-04-02 at 02 40 23

Let me know your thoughts.

### CODE REVIEW

**Comment by turbo124** [comment]
@Civolilah 

I think the dropdown button is the best UX here, also, i would change the text to

Save / Send Now / Start

**Comment by Civolilah** [comment]
> @Civolilah
> 
> I think the dropdown button is the best UX here, also, i would change the text to
> 
> Save / Send Now / Start

@turbo124 Excellent, just updated PR with additional changes. Let me know your thoughts.

File: src/pages/recurring-invoices/common/hooks.tsx [hunk 1]
```diff
@@ -200,13 +200,22 @@ export function useSave(props: RecurringInvoiceSaveProps) {
...
+    queryAction?: 'send_now' | 'start'
+  ) => {
     toast.processing();
     setErrors(undefined);
 
+    let endpointUrl = '/api/v1/recurring_invoices/:id';
+
+    if (queryAction) {
+      endpointUrl = '/api/v1/recurring_invoices/:id?' + queryAction + '=true';
+    }
```
[the remark this answers is not in the document]
Comment (Civolilah) [reply src/pages/recurring-invoices/common/hooks.tsx]: Changed!

File: src/pages/recurring-invoices/common/hooks.tsx [hunk 2]
```diff
@@ -443,11 +452,20 @@ export function useActions() {
...
+  ) => {
     setErrors(undefined);
     toast.processing();
 
-    request('POST', endpoint('/api/v1/recurring_invoices'), recurringInvoice)
+    let endpointUrl = '/api/v1/recurring_invoices';
+
+    if (queryAction) {
+      endpointUrl = '/api/v1/recurring_invoices?' + queryAction + '=true';
+    }
```
[the remark this answers is not in the document]
Comment (Civolilah) [reply src/pages/recurring-invoices/common/hooks.tsx]: Changed!

**Review by beganovich — APPROVED** [review]

**Pull request merged**